src/wpanctl/tool-cmd-peek.c

```diff
@@ -0,0 +1,319 @@
+/*
+ *
+ * Copyright (c) 2017 Nest Labs, Inc.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <getopt.h>
+#include "wpanctl-utils.h"
+#include "tool-cmd-peek.h"
+#include "assert-macros.h"
+#include "wpan-dbus-v1.h"
+#include "args.h"
+
+#include <errno.h>
+
+const char peek_cmd_syntax[] = "[args] <address>";
+
+static const arg_list_item_t peek_option_list[] = {
+	{'h', "help", NULL, "Print Help"},
+	{'t', "timeout", "ms", "Set timeout period"},
+	{'c', "count", "bytes", "Number of bytes to peek"},
+	{'d', "data", "", "Show the result as a sequence of unformatted hex bytes"},
+	{0}
+};
+
+
+static void dump_data_line(const uint8_t *data_ptr, uint16_t data_len)
+{
+	char buf[100];
+	char *cur = buf;
+	uint16_t i;
+
+	*cur++ = '|';
+
+	for (i = 0; i < 16; i++) {
+		*cur++ = ' ';
+		if (i < data_len) {
+			cur += sprintf(cur, "%02X", data_ptr[i]);
+		} else {
+			*cur++ = ' ';
+			*cur++ = ' ';
+		}
+
+		if ((i == 7) || (i == 15)) {
+			*cur++ = ' ';
+			*cur++ = '|';
+		}
+	}
+
+	*cur++ = ' ';
+
+	for (i = 0; i < 16; i++)
+	{
+		char c = (char)(0x7f & (data_ptr)[i]);
+
+		if (i < data_len) {
+			*cur++ = isprint(c) ? c : '.';
+		} else {
+			*cur++ = ' ';
+		}
+	}
+
+	*cur++ = ' ';
+	*cur++ = '|';
+
+	*cur = '\0';
+
+	fprintf(stderr, "%s\n", buf);
+}
+
+static void dump_data(const uint8_t *data_ptr, uint16_t data_len)
+{
+	uint16_t len;
+	char buf[100];
+	char *cur;
+	uint16_t address_index = 0;
+	uint16_t i;
+
+	fprintf(stderr, "+------+-------------------------+-------------------------+------------------+\n");
+
+	while (data_len > 0)
```

Comment:
Yes. I actually included the `dump_data_line()` in the loop itself (but forgot to delete it)...
Will fix it... :)